examples/Catena4430_Sensor/Catena4430_cMeasurementLoop_SDcard.cpp

```diff
@@ -571,4 +575,75 @@ cMeasurementLoop::updateFromSd(
         }
     }
 
+void
+cMeasurementLoop::handleSdTTNv3Migrate(
+    void
+    )
+    {
+    static const char * const sMigrate = "MIGRATE.V3";
+    bool fMigrate;
+    bool fResult = this->checkSdCard();
+
+	if (fResult)
+        {
+        fMigrate = this->handleSdTTNv3MigrateCardUp(sMigrate);
+        }
+
+    if (fMigrate)
```

Comment:
If `fResult` if false, `fMigrate` is uninitialized.

This logic is awfully convoluted. It would be clearer to have all  these simple decisions in one function.

Reply:
have initialized `fMigrate` with `false`. I would like to get clear on this comment.

> It would be clearer to have all these simple decisions in one function.
- whether this means having a init function for the flags?

Reply:
When I looked at the code, `fMigrate` was not initialized. It was declared just above without an initializer (line 584). THis is still the case. Therefore, if the branch at line 588 is not taken, 589 will not be executed, and 592 will have undefined behavior.

My other comment is that there are too many functions here. One function to handle migration would be enough.

Reply:
Thank you Terry. I have handled migration of device with one function. However I have used different function to rejoin network, as that might be used for any other requirement in future.